src/python/pants/backend/jvm/subsystems/zinc.py

```diff
@@ -210,10 +210,12 @@ def dist(self):
         get_buildroot())
 
       # Since this code can be run in multi-threading mode due to multiple
-      # zinc workers, we need to make sure the file operations below is atomic.
+      # zinc workers, we need to make sure the file operations below are atomic.
       with self._lock:
-        # Create the symlink if it does not exist
+        # Create the symlink if it does not exist (or points to a file that doesn't exist,
+        # e.g., a JDK that is no longer present).
         if not os.path.exists(jdk_home_symlink):
+          safe_delete(jdk_home_symlink)  # In case it's a broken symlink.
```

Comment:
These branches are now the same, so could be unified into a single if branch if you fancied it. Or not :)

Reply:
Done.